### PR TITLE
Template and e2e test for deleted notification

### PR DESCRIPTION
**Pull request by victorpras**
Template and e2e test for deleted notification

### CODE REVIEW

File: e2e/testcases/notification_test.go [hunk 1]
```diff
@@ -567,6 +567,73 @@ func TestNotificationOnSyncCreated(t *testing.T) {
...
+	err = nt.Watcher.WatchObject(kinds.Deployment(), rootReconcilerName, rootSyncNN.Namespace,
+		[]testpredicates.Predicate{testpredicates.DeploymentHasContainer(reconcilermanager.Notification)})
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+
+	// deleting root-sync-3
+	nt.Must(nt.RootRepos[configsync.RootSyncName].Remove(nomostest.StructuredNSPath(rootSyncNN.Namespace, rootSyncNN.Name)))
+	nt.Must(nt.RootRepos[configsync.RootSyncName].CommitAndPush("Remove root-sync-3"))
+	tg := taskgroup.New()
```
Comment (sdowell) [inline e2e/testcases/notification_test.go:613]: No need for a taskgroup here, as there's only a single thing we're waiting for

Comment (victorpras) [reply e2e/testcases/notification_test.go]: removed the taskgroup

**Review by sdowell — APPROVED** [review]

**Pull request merged**